### ISSUE TITLE
Use CMake to install executables

### ISSUE THREAD

**Issue by EdwardSafford-NOAA** [NOAA-EMC/GSI-Monitor#5]
At present the CMake scripts dump all executables in the the GSI-monitor/install/bin directory.  Modify the scripts to place the executables in the various exec directories within each monitor.

Note that the placement and number of exec directories can be reduced/streamlined.  If that is undertaken it will be under a separate issue.

**Comment by EdwardSafford-NOAA** [comment]
@DavidHuber-NOAA I wonder if you might have time to test building the GSI-monitor on S4 and/or jet.  If so, you'll need to checkout the GSI-monitor/feature/Install_5 branch.  If not I understand.

Note that GSI-monitor has been removed from the GSI repo and contains all the monitoring packages.  But there is still a dependency on ncdiag which (for the moment) remains in the GSI package.  So you'll have to do a minimal GSI build and then set ncdiag_DIR = /path/to/GSI/install/lib64/cmake/ncdiag before running the GSI-monitor/ush/build.sh script.

**Comment by DavidHuber-NOAA** [comment]
@EdwardSafford-NOAA Sure, I will give it a go on both systems.  I should be able to get to it this afternoon or perhaps tomorrow.  Thanks for the heads up.

**Comment by EdwardSafford-NOAA** [comment]
@DavidHuber-NOAA  Super!  Thanks very much.

**Comment by DavidHuber-NOAA** [comment]
@EdwardSafford-NOAA I was able to build on Jet after making a few changes to the `jet.lua` modulefile.  Since NetCDF is available on Jet only after loading the `hpc-impi` module, I had to add that.  Similarly, I loaded a newer CMake module.

S4 is currently down due to a hardware issue, so I may not be able to test build there until Tuesday.

**Comment by DavidHuber-NOAA** [comment]
Ah, I see now that I tried building from the wrong branch on Jet.  Let me give that another go.

**Comment by DavidHuber-NOAA** [comment]
@EdwardSafford-NOAA OK, the build was successful the first time around this time.

**Comment by EdwardSafford-NOAA** [comment]
@DavidHuber-NOAA Excellent!  Thanks.

**Comment by aerorahul** [comment]
@EdwardSafford-NOAA 
Can you explain this issue a bit more?
Installation of binaries should not be in the cloned space.  We currently do this and are moving away from this non-standard practice.

**Comment by EdwardSafford-NOAA** [comment]
@aerorahul Ok, my ignorance on the policy direction.  Should binaries remain in the top level install/bin directory?  

I'll change this issue to modify the scripts that use binaries to find them in the new location.

**Comment by aerorahul** [comment]
The scripts should look for the executables in the `CMAKE_INSTALL_PREFIX/bin` directory. 
Currently, we ignorantly set `CMAKE_INSTALL_PREFIX=SOURCE_DIR`

At runtime, `CMAKE_INSTALL_PREFIX` will be unavailable, instead we use `SOURCE_DIR`.
I don't know how the scripts in this repo are set up, so may be I need to understand that a little more before I can give you any concrete suggestions.

**Comment by EdwardSafford-NOAA** [comment]
At present the scripts and executables are located by function (data extraction, utility, image generation) within each monitor.  The monitor scripts use a monitor level configuration script to find the executables.  That config script is set up by the perl install scripts. 

My initial thought is to merge those 4 monitor specific perl install scripts into a single script and run it as part of the cmake install.

**Comment by DavidHuber-NOAA** [comment]
@EdwardSafford-NOAA The monitors built successfully on S4 as well.

**Comment by EdwardSafford-NOAA** [comment]
Closing this issue.  Will open a replacement issue to update scripts to find execs rather than altering cmake install.   The build script constructed in this issue was added to #7 and merged under PR #8.
